packages/vsce/tests/SshConfigUtils.test.ts

```diff
@@ -0,0 +1,383 @@
+                expect(result).toBe("Option 1");
+            });
+        });
+        describe("showCustomMenu", () => {
+            beforeEach(() => {
+                vi.spyOn(vscode.window, "createQuickPick").mockReturnValue(mockQuickPick);
+            });
+            afterEach(() => {
+                vi.restoreAllMocks();
+            });
+            it("shows mapped items in quick pick", async () => {
+                const opts = {
+                    title: "Select SSH",
+                    placeholder: "Pick one",
+                    items: [{ label: "host1", description: "host" }],
+                };
+                const promise = testInstance.callShowCustomMenu(opts);
+
+                expect(mockQuickPick.items).toEqual([{ label: "host1", description: "host" }]);
+                expect(mockQuickPick.title).toBe("Select SSH");
+                expect(mockQuickPick.placeholder).toBe("Pick one");
+                expect(mockQuickPick.ignoreFocusOut).toBe(true);
+
+                const onDidHide = (mockQuickPick.onDidHide as any).mock.calls[0][0];
+                onDidHide();
+
+                const result = await promise;
+                expect(result).toBeUndefined();
+            });
+            it("returns custom item when input starts with >", async () => {
+                const opts = { items: [{ label: "host1", description: "host" }] };
+
+                const promise = testInstance.callShowCustomMenu(opts);
+                const onDidChangeValue = (mockQuickPick.onDidChangeValue as any).mock.calls[0][0];
+                onDidChangeValue("customHost");
+
+                expect(mockQuickPick.items[0]).toEqual({
+                    label: "> customHost",
+                    description: "Custom SSH Host",
+                });
+                mockQuickPick.selectedItems = [mockQuickPick.items[0]];
+                const onDidAccept = (mockQuickPick.onDidAccept as any).mock.calls[0][0];
+                onDidAccept();
+
+                const result = await promise;
+                expect(result).toEqual({
+                    label: "customHost",
+                    description: "Custom SSH Host",
+                });
+            });
+
+            it("returns undefined when quick pick is hidden", async () => {
+                const opts = { items: [] };
+                const promise = testInstance.callShowCustomMenu(opts);
+                const onDidHide = (mockQuickPick.onDidHide as any).mock.calls[0][0];
+                onDidHide();
+                const result = await promise;
+                expect(result).toBeUndefined();
+            });
+        });
+        describe("getCurrentDir", () => {
+            it("returns the fsPath when workspaceRoot is defined", () => {
+                const testPath = "/user/project";
+                (ZoweVsCodeExtension as any).workspaceRoot = {
+                    uri: { fsPath: testPath },
+                };
```

Comment:
Thoughts on using `jest.replaceProperty` here instead?

If you run into issues using it, feel free to disregard ^^